### PR TITLE
prune README

### PR DESCRIPTION
**Pull request by esc**
Remove all "redundent" information from the README and place all the
badges at the top.

I guess, the depth of the README is a matter of choice? But perhaps
avoiding duplication is a goal worth pursuing?

### CODE REVIEW

**Comment by esc** [comment]
Here is how it renders: https://github.com/esc/numba/blob/beautify_readme/README.rst

**Comment by esc** [comment]
Consider this a `discussion`.

**Comment by stuartarchibald** [comment]
Please could you resolve the conflicts @esc? IIRC the OOB consensus on this was to merge as it reduces the number of places where dependencies are listed, this preventing potential divergence as updates are made.

**Comment by sklam** [comment]
I think the readme should have a direct link to the installation instruction and a link to the demo notebooks.

**Comment by esc** [comment]
I rebased the conflict away, the Azure badge bow points at `main`.

File: README.rst [hunk 1]
```diff
@@ -38,6 +38,19 @@ For more information about Numba, see the Numba homepage:
...
+Please follow the instuctions:
+
+* https://numba.readthedocs.io/en/stable/user/installing.html
+
+Demo
+====
+
+Please have a look and the demo notebooks:
+
+* https://github.com/numba/numba-examples/tree/master/notebooks
```
[the remark this answers is not in the document]
Comment (esc) [reply README.rst]: OK, I have inserted the mybinder link instead.

**Comment by esc** [comment]
FWIW: llvmlite dependencies are now documented in the Sphinx docs as part of this PR:

https://github.com/numba/numba/pull/7833#pullrequestreview-885302611

**Pull request merged**